### PR TITLE
Enable nullable reference types in `Ical.Net.Tests.csproj` 

### LINKED ISSUE #904: chore: Update unit tests with project-wide `nullable enable`

### PR DESCRIPTION
**Pull request by axunonb**
No changes in code logic, only affecting unit tests.

* Add `nullable enable` to `Ical.Net.Tests`
* Fix nullability warnings by applying nullable and null-forgiving operators in unit tests
* Replace `Assert.Multiple` with `using (Assert.EnterMultipleScope())`

Resolves #904

### CODE REVIEW

File: Ical.Net.Tests/Logging/Abstractions/InMemorySink.cs [hunk 1]
```diff
@@ -14,13 +13,14 @@ namespace Ical.Net.Tests.Logging.Abstractions;
 
 internal class InMemorySink : ILogEventSink
 {
-    public static InMemorySink Instance;
+    public static InMemorySink Instance = null!;
 
-    private ConcurrentQueue<Serilog.Events.LogEvent> _events { get; } = new();
+    private ConcurrentQueue<Serilog.Events.LogEvent> Events { get; } = new();
```
Comment (maknapp) [inline Ical.Net.Tests/Logging/Abstractions/InMemorySink.cs]: This should be a readonly field not a property, and probably named back to `_events`.
```c#
private readonly ConcurrentQueue<Serilog.Events.LogEvent> _events = new();
```

File: Ical.Net.Tests/WikiSamples/RecurrenceWikiTests.cs [hunk 2]
```diff
@@ -129,12 +127,12 @@ public void DailyIntervalCount()
...
         var calendarSerializer = new CalendarSerializer();
-        var generatedIcs = calendarSerializer.SerializeToString(calendar);
+        var generatedIcs = calendarSerializer.SerializeToString(calendar)!;
 
         // Calculate all occurrences
         var tz = TimeZoneResolvers.Default("Europe/Zurich");
-        IEnumerable<Occurrence> occurrences = calendar.GetOccurrences(tz);
-        Assert.That(occurrences.Count(), Is.EqualTo(2));
+        IEnumerable<Occurrence> occurrences = calendar.GetOccurrences(tz).ToList();
+        Assert.That(occurrences.Count, Is.EqualTo(2));
```
Comment (maknapp) [inline Ical.Net.Tests/WikiSamples/RecurrenceWikiTests.cs]: This is still using `IEnumerable.Count()` instead of `List<T>.Count`. Change to `var occurrences = ...`, and `Has.Count.EqualTo`.

File: Ical.Net.Tests/DataTypeTest.cs [hunk 3]
```diff
@@ -14,15 +14,13 @@ public class DataTypeTest
 {
     [Test, Category("DataType")]
     public void OrganizerConstructorMustAcceptNull()
-    {
-        Assert.DoesNotThrow(() => { var o = new Organizer(null); });
-    }
+        => Assert.DoesNotThrow(() => { var o = new Organizer(null!); });
```
Comment (maknapp) [inline Ical.Net.Tests/DataTypeTest.cs]: Why does `Organizer` not accept nullable `string?` if it allows for null values? Same for `Attachment(string)` below.

File: Ical.Net.Tests/SerializationTests.cs [hunk 4]
```diff
@@ -46,7 +46,7 @@ internal static void CompareComponents(ICalendarComponent cb1, ICalendarComponen
                 }
                 else if (p1.Value is IEnumerable)
                 {
-                    CompareEnumerables((IEnumerable)p1.Value, (IEnumerable)p2.Value, p1.Name);
+                    CompareEnumerables((IEnumerable)p1.Value, (IEnumerable) p2.Value!, p1.Name);
```
Comment (maknapp) [inline Ical.Net.Tests/SerializationTests.cs]: In this PR or another, `CompareEnumerables()` could be replaced with `Assert.That(..., Is.EquivalentTo(...))`.

File: Ical.Net.Tests/SymmetricSerializationTests.cs [hunk 5]
```diff
@@ -208,23 +205,22 @@ public void CategoriesTest(string[] categories)
 
     private static IEnumerable CategoryTest_TestCases()
     {
-        yield return new string[] { "Foo", "Bar", "Baz" };
-        yield return new string[] { "Hello", "world" };
-        yield return new string[] { "Hello", "world", null };
+        yield return new[] { "Foo", "Bar", "Baz" };
+        yield return new[] { "Hello", "world" };
```
Comment (maknapp) [inline Ical.Net.Tests/SymmetricSerializationTests.cs:209]: No need to test `null` anymore?

**Review by maknapp — CHANGES_REQUESTED** [review]

File: Ical.Net.Tests/WikiSamples/RecurrenceWikiTests.cs (hunk 2 above)
Comment (axunonb) [reply Ical.Net.Tests/WikiSamples/RecurrenceWikiTests.cs]: oops, tnx - of course

File: Ical.Net.Tests/Logging/Abstractions/InMemorySink.cs (hunk 1 above)
Comment (axunonb) [reply Ical.Net.Tests/Logging/Abstractions/InMemorySink.cs]: 👍

File: Ical.Net.Tests/DataTypeTest.cs (hunk 3 above)
Comment (axunonb) [reply Ical.Net.Tests/DataTypeTest.cs]: `Organizer` and `Attachment` constructors did not accept `null`, although there was a check for a `null` argumenta. Existing code might pass `null` though.
I changed the constructors' signature for now.

File: Ical.Net.Tests/SymmetricSerializationTests.cs (hunk 5 above)
Comment (axunonb) [reply Ical.Net.Tests/SymmetricSerializationTests.cs:209]: No, indeed not, because `CalendarEvent.Categories` is an `IList<string>`

File: Ical.Net.Tests/SerializationTests.cs (hunk 4 above)
Comment (axunonb) [reply Ical.Net.Tests/SerializationTests.cs]: Indeed, this was the only reference, so no problem.

**Comment by axunonb** [comment]
Thanks for review!

File: Ical.Net.Tests/SerializationTests.cs [hunk 6]
```diff
@@ -44,9 +44,9 @@ internal static void CompareComponents(ICalendarComponent cb1, ICalendarComponen
                     if (((IComparable)p1.Value).CompareTo(p2.Value) != 0)
                         continue;
                 }
-                else if (p1.Value is IEnumerable)
+                else if (p1.Value is IEnumerable p1Enumerable)
                 {
-                    CompareEnumerables((IEnumerable)p1.Value, (IEnumerable) p2.Value!, p1.Name);
+                    Assert.That(p1Enumerable, Is.EquivalentTo((IEnumerable) p2.Value!));
```
Comment (maknapp) [inline Ical.Net.Tests/SerializationTests.cs]: Add `Assert.That(..., message: p1.Name + " do not match");` to match the error message from before.

**Review by maknapp — CHANGES_REQUESTED** [review]

File: Ical.Net.Tests/SerializationTests.cs (hunk 6 above)
Comment (axunonb) [reply Ical.Net.Tests/SerializationTests.cs]: Done

**Review by maknapp — APPROVED** [review]
